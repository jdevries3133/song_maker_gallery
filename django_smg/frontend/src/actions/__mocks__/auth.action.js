import { LOGIN, REGISTER, CLEAR_ERROR, LOGOUT, UPDATE, DELETE } from "../types";

export const register = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: REGISTER,
    payload: {
      isAuthenticated: true,
      authError: false,
      token: "testtoken",
      user: "testuser",
    },
  });
});

export const clearError = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: CLEAR_ERROR,
    payload: {
      authError: false,
    },
  });
});

export const logout = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: LOGOUT,
  });
});

export const tryToken = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: LOGIN,
    payload: {
      isAuthenticated: true,
      authError: false,
      token: "testtoken",
    },
  });
});

export const login = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: LOGIN,
    payload: {
      isAuthenticated: true,
      authError: false,
      token: "testtoken",
    },
  });
});

export const updateUser = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: UPDATE,
    payload: {
      authError: false,
      token: "testtoken",
    },
  });
});

export const deleteAccount = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: DELETE,
    payload: {
      isAuthenticated: false,
      token: null,
    },
  });
});
