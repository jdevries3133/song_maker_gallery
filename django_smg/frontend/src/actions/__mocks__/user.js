import { POST_GALLERY, GET_GALLERIES, DELETE_GALLERY } from "../types";

// POST GALLERY
export const postGallery = jest
  .fn()
  .mockImplementation((form) => (dispatch) => {
    dispatch({
      type: POST_GALLERY,
      payload: {
        status: 201,
        formPassthrough: null,
        gallery: {
          title: form.title,
          description: form.description,
          slug: form.title.toLowerCase().replace(" ", "-"),
          pk: 1,
        },
      },
    });
  });

// GET_GALLERIES
export const getUserGalleries = jest
  .fn()
  .mockImplementation(() => (dispatch) => {
    dispatch({
      type: GET_GALLERIES,
      payload: {
        galleries: [
          {
            title: "e",
            description:
              "We will always find a way to share music. In lieu of the concert hall, our musical performances today are expressed in ones and zeroes, but they are none the less as human and as meaningful as always.\n\nPlease enjoy this showcase of our school's music lab compositions. Our students' creativity truly knows no bounds!",
            slug: "e",
            pk: 1,
          },
          {
            title: "e",
            description:
              "We will always find a way to share music. In lieu of the concert hall, our musical performances today are expressed in ones and zeroes, but they are none the less as human and as meaningful as always.\n\nPlease enjoy this showcase of our school's music lab compositions. Our students' creativity truly knows no bounds!",
            slug: "e-1",
            pk: 2,
          },
          {
            title: "e",
            description:
              "We will always find a way to share music. In lieu of the concert hall, our musical performances today are expressed in ones and zeroes, but they are none the less as human and as meaningful as always.\n\nPlease enjoy this showcase of our school's music lab compositions. Our students' creativity truly knows no bounds!",
            slug: "e-2",
            pk: 3,
          },
        ],
      },
    });
  });

// DELETE_GALLERY
export const deleteGallery = jest.fn().mockImplementation(() => (dispatch) => {
  dispatch({
    type: DELETE_GALLERY,
    payload: { status: "deleted" },
  });
});

/* Must remove "deleted" status from redux state after user acknowledgement */
export const acknowledgeDelete = jest
  .fn()
  .mockImplementation(() => (dispatch) => {
    dispatch({
      type: DELETE_GALLERY,
      payload: { status: "delete_acknowledged" },
    });
  });
