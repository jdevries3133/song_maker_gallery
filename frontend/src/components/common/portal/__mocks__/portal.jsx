export const Portal = jest.fn().mockImplementation(({ children }) => children);
