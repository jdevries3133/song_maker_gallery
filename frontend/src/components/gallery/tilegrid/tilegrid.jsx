import React from "react";
import { DynamicTile } from "Common/song_tiles";
import useWidth from "Common/useWidth";

import styled, { A, P, Button, Div, H2 } from "Styles";

const Ul = styled.ul`
  text-align: center;
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

const GalleryContainer = styled(Div)`
  margin-top: 8rem;
`;

const TgTitle = styled(H2)`
  margin: 0 0 2rem 0;
  position: relative;
  z-index: 10;
  top: min(10vw, 5.9rem);
  font-size: min(6vw, 3.5rem);
  color: black;
  background-color: white;
  border-radius: 20px;
  box-shadow: 0px 3px 8px rgb(100, 100, 100);
  padding: 7px;
  transition: 0.4s ease;

  @media (max-width: 600px) {
    position: static;
    font-size: 2rem;
    border-radius: 15px;
  }

  @media (min-width: 600px) {
    &:hover {
      cursor: pointer;
      background: green;
      opacity: 0%;
    }

    &:hover ~ span {
      opacity: 100%;
      transform: rotate(180deg);
    }
  }
`;

const Finger = styled.span`
  position: relative;
  bottom: max(1.3vw, 1.7rem);
  opacity: 0%;
  color: white;
  display: inline-block;
  background-color: #5a38cc;
  border-radius: 20px;
  font-size: min(7vw, 3.5rem);
  transform: rotate(0deg);
  transition: 0.4s ease;

  @media (max-width: 600px) {
    position: static;
    background: none;
    color: black;
    opacity: 100%;
    font-family: "Noto Sans JP", sans-serif;
    font-size: 4rem;
    padding: 0;
    transform: rotate(180deg);
  }
`;

export const TileGrid = ({ data }) => {
  const { width } = useWidth(600);

  return width > 600 ? (
    <GalleryContainer id="gallery">
      <a href="#gallery_top">
        <TgTitle id={data.group_name}>{data.group_name}</TgTitle>
        <br />
        <Finger>☟</Finger>
      </a>
      <Ul>
        {data.songs.map((song, index) => (
          <DynamicTile song={song} pixelWidth={300} key={song.songId + index} />
        ))}
      </Ul>
    </GalleryContainer>
  ) : (
    <GalleryContainer id="gallery">
      <TgTitle id={data.group_name}>{data.group_name}</TgTitle>
      <A href="#gallery_top">
        <Button block>
          <Finger>☟</Finger>
          <P>Top</P>
        </Button>
      </A>
      <div style={{ textAlign: "center" }}>
        <Ul>
          {data.songs.map((song, index) => (
            <DynamicTile
              mobile={true}
              song={song}
              pixelWidth={300}
              key={song.songId + index}
            />
          ))}
        </Ul>
      </div>
    </GalleryContainer>
  );
};
