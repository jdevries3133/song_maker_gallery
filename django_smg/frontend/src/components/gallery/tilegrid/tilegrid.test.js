import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { TileGrid } from "./tilegrid";

import { apiResponse } from "Test/__mocks__/api_response";

test("tilegrid matches snapshot", () => {
  const tree = renderer
    .create(<TileGrid data={apiResponse.song_groups[0]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
