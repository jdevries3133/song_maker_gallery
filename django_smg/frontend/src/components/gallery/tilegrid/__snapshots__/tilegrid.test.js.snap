// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  text-align: center;
  padding: inherit;
}

.c2 {
  display: inline-block;
  background-color: "#88ff00";
  font-weight: 400;
  padding: 30px 0;
  border-radius: 20px;
}

.c6 {
  display: inline-block;
  text-align: left;
}

.c6:nth-child(7n + 0):hover > * > * > svg {
  background-color: #e33059;
}

.c6:nth-child(7n + 1):hover > * > * > svg {
  background-color: #f7943e;
}

.c6:nth-child(7n + 2):hover > * > * > svg {
  background-color: #ecd92a;
}

.c6:nth-child(7n + 3):hover > * > * > svg {
  background-color: #95c632;
}

.c6:nth-child(7n + 4):hover > * > * > svg {
  background-color: #12826d;
}

.c6:nth-child(7n + 5):hover > * > * > svg {
  background-color: #5b38cc;
}

.c6:nth-child(7n + 6):hover > * > * > svg {
  background-color: #ea57b2;
}

.c8 {
  border-radius: 10px;
  -webkit-transition: 100ms;
  transition: 100ms;
}

.c8:hover > rect {
  opacity: 0;
  stroke: var(--purple);
  stroke-width: 8px;
  -webkit-transition: 100ms;
  transition: 100ms;
}

.c8:hover > text {
  opacity: 100;
}

.c9 {
  font: bold 40px "Indie Flower";
  fill: white;
  -webkit-transition: 100ms;
  transition: 100ms;
  opacity: 0;
}

.c7 {
  width: 300px;
  height: 103px;
  display: inline-block;
  background-color: white;
  box-shadow: 5px 5px 8px #ddd;
  margin: 10px;
}

.c7:hover {
  border-radius: 20px;
}

.c5 {
  text-align: center;
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c1 {
  margin-top: 8rem;
}

.c3 {
  margin: 0 0 2rem 0;
  position: relative;
  z-index: 10;
  top: min(10vw,5.9rem);
  font-size: min(6vw,3.5rem);
  color: black;
  background-color: white;
  border-radius: 20px;
  box-shadow: 0px 3px 8px rgb(100,100,100);
  padding: 7px;
  -webkit-transition: 0.4s ease;
  transition: 0.4s ease;
}

.c4 {
  position: relative;
  bottom: max(1.3vw,1.7rem);
  opacity: 0%;
  color: white;
  display: inline-block;
  background-color: #5a38cc;
  border-radius: 20px;
  font-size: min(7vw,3.5rem);
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.4s ease;
  transition: 0.4s ease;
}

@media (max-width:600px) {
  .c3 {
    position: static;
    font-size: 2rem;
    border-radius: 15px;
  }
}

@media (min-width:600px) {
  .c3:hover {
    cursor: pointer;
    background: green;
    opacity: 0%;
  }

  .c3:hover ~ span {
    opacity: 100%;
    -webkit-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }
}

@media (max-width:600px) {
  .c4 {
    background: none;
    color: black;
    opacity: 100%;
    font-family: "Noto Sans JP",sans-serif;
    font-size: 4rem;
    padding: 0;
    -webkit-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }
}

<div
  className="c0 c1"
  id="gallery"
>
  <a
    href="#gallery_top"
  >
    <h2
      className="c2 c3"
      id={
        Array [
          "A Group of Marks",
        ]
      }
    >
      A Group of Marks
    </h2>
    <br />
    <span
      className="c4"
    >
      â˜Ÿ
    </span>
  </a>
  <ul
    className="c5"
  >
    <div
      className="c0 c6"
    >
      <div
        className="c0 c7"
        height="103px"
        width="300px"
      >
        <a
          href="https://musiclab.chromeexperiments.com/Song-Maker/song/5676759593254912"
          rel="noopner noreferrer"
          target="_blank"
        >
          <svg
            className="c8"
            viewBox="0 0 300 103"
            xmlns="https://www.w3.org/2000/svg"
          >
            <text
              className="c9"
              dominantBaseline="middle"
              textAnchor="middle"
              x="50%"
              y="50%"
            >
              Mark J.
            </text>
          </svg>
        </a>
      </div>
    </div>
    <div
      className="c0 c6"
    >
      <div
        className="c0 c7"
        height="103px"
        width="300px"
      >
        <a
          href="https://musiclab.chromeexperiments.com/Song-Maker/song/5676759593254912"
          rel="noopner noreferrer"
          target="_blank"
        >
          <svg
            className="c8"
            viewBox="0 0 300 103"
            xmlns="https://www.w3.org/2000/svg"
          >
            <text
              className="c9"
              dominantBaseline="middle"
              textAnchor="middle"
              x="50%"
              y="50%"
            >
              Mark J.
            </text>
          </svg>
        </a>
      </div>
    </div>
    <div
      className="c0 c6"
    >
      <div
        className="c0 c7"
        height="103px"
        width="300px"
      >
        <a
          href="https://musiclab.chromeexperiments.com/Song-Maker/song/5676759593254912"
          rel="noopner noreferrer"
          target="_blank"
        >
          <svg
            className="c8"
            viewBox="0 0 300 103"
            xmlns="https://www.w3.org/2000/svg"
          >
            <text
              className="c9"
              dominantBaseline="middle"
              textAnchor="middle"
              x="50%"
              y="50%"
            >
              Mark J.
            </text>
          </svg>
        </a>
      </div>
    </div>
    <div
      className="c0 c6"
    >
      <div
        className="c0 c7"
        height="103px"
        width="300px"
      >
        <a
          href="https://musiclab.chromeexperiments.com/Song-Maker/song/5676759593254912"
          rel="noopner noreferrer"
          target="_blank"
        >
          <svg
            className="c8"
            viewBox="0 0 300 103"
            xmlns="https://www.w3.org/2000/svg"
          >
            <text
              className="c9"
              dominantBaseline="middle"
              textAnchor="middle"
              x="50%"
              y="50%"
            >
              Mark J.
            </text>
          </svg>
        </a>
      </div>
    </div>
  </ul>
</div>
`;
